train_supervised.py

```diff
@@ -1,20 +1,21 @@
+import json
+import os
+
+import numpy as np
+import replicate
 import torch
-import torch.optim as optim
 import torch.nn as nn
+import torch.optim as optim
 import torchvision
 import torchvision.transforms as transforms
-import numpy as np
 from tqdm import tqdm
-import json
-import os
-import replicate
+
 from models import wide_resnet, resnet_official
 
 dir_path = os.path.dirname(os.path.realpath(__file__))
 p = os.path.join(dir_path, 'params.json')
 with open(p, 'r') as f:
     params = json.load(f)
-    f.close()
```

Comment:
The `with` statement means you don't have to close the file here.  Once the code is out of the `with` block, python closes it for you.  

Technically what's going on here is that the file object returned from `open()` is a [Context Manager](https://docs.python.org/3/reference/datamodel.html#context-managers).  (No need to read and understand the docs at that link, just if you're curious.)   Context Managers are a way of defining a safe little runtime for code that might fail.